File: SQL/Select/SQLSelectTable.vb

' ___________________________________________________
'
'  © Hi-Integrity Systems 2010. All rights reserved.
'  www.hisystems.com.au - Toby Wicks
' ___________________________________________________
'

Option Strict On
Option Explicit On

Namespace SQL

    Public Class SQLSelectTable
        Inherits SQLSelectTableBase

        Private pstrDatabaseName As String
        Private pstrSchemaName As String
        Private pstrName As String

        Public Sub New()

        End Sub

        Public Sub New(ByVal strName As String)

            Me.Name = strName

        End Sub

        Public Sub New(ByVal strName As String, ByVal strAlias As String)

            Me.New(strName)
            MyBase.Alias = strAlias

        End Sub

        Public Property DatabaseName As String
            Get

                Return pstrDatabaseName

            End Get

            Set(value As String)

                pstrDatabaseName = value

            End Set
        End Property

        Public Property SchemaName As String
            Get

                Return pstrSchemaName

            End Get

            Set(ByVal value As String)

                pstrSchemaName = value

            End Set
        End Property

        Public Property Name() As String
            Get

                Return pstrName

            End Get

            Set(ByVal Value As String)

                If Value.Trim = String.Empty Then
                    Throw New ArgumentNullException
                Else
                    pstrName = Value
                End If

            End Set
        End Property

        Protected Overrides ReadOnly Property Source(ByVal eConnectionType As Database.ConnectionType) As String
            Get

                If Me.Name.Trim = String.Empty Then
                    Throw New Exceptions.DatabaseObjectsException("Table has not been set.")
                End If

                Dim strSQL As String = String.Empty

                If pstrDatabaseName <> String.Empty Then
                    strSQL &= SQLConvertIdentifierName(pstrDatabaseName, eConnectionType) & "."
                End If

                If pstrSchemaName <> String.Empty Then
                    strSQL &= SQLConvertIdentifierName(pstrSchemaName, eConnectionType) & "."
                End If

                Return strSQL & SQLConvertIdentifierName(Me.Name, eConnectionType)

            End Get
        End Property

    End Class

End Namespace

File: SQL/Select/SQLSelectTable.cs
// _________________________________________________________________________
//
//  © Hi-Integrity Systems 2010. All rights reserved.
//  www.hisystems.com.au - Toby Wicks
// 
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//	    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// _________________________________________________________________________
//

using System.Collections;
using System;
using System.Data;

namespace DatabaseObjects.SQL
{
	public class SQLSelectTable : SQLSelectTableBase
	{
		private string pstrDatabaseName;
		private string pstrSchemaName;
		private string pstrName;
			
		public SQLSelectTable()
		{
		}
			
		public SQLSelectTable(string strName)
		{
			this.Name = strName;
		}
			
		public SQLSelectTable(string strName, string strAlias)
            : this(strName)
		{
			base.Alias = strAlias;
		}
			
		public string DatabaseName
		{
			get
			{
				return pstrDatabaseName;
			}
				
			set
			{
				pstrDatabaseName = value;
			}
		}
			
		public string SchemaName
		{
			get
			{
				return pstrSchemaName;
			}
				
			set
			{
				pstrSchemaName = value;
			}
		}
			
		public string Name
		{
			get
			{
				return pstrName;
			}
				
			set
			{
				if (String.IsNullOrEmpty(value))
					throw new ArgumentNullException();
				
				pstrName = value;
			}
		}

		internal override string Source(Serializers.Serializer serializer)
		{
			return serializer.SerializeSelectTable(this);
		}
	}
}
